### PR TITLE
Kuc unit test date time

### PR DESCRIPTION
**Pull request by minhhy2801**

### CODE REVIEW

File: src/react/DateTime/__test__/onEvent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        timeFormat="dd:aa"
+        onChange={onChange}
+      />
+    );
+    const node = container.getElementsByClassName('kuc-input-text text-input')[0];
+    fireEvent.click(node, {target: {value: '04/10/2020'}});
+  });
+  test('change value DateTime', () => {
+    const onChange = (value: Date) => {
+      expect(true);
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/onEvent.test.tsx]: Should expect `value` to be correct.

File: src/react/DateTime/__test__/render.test.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  afterEach(() => {
+    // @ts-ignore
+    console.error.mockRestore();
+  });
+
+  test('render without props DateTime', () => {
+    const {container} = render(<DateTime />);
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('date-time-container');
+    }
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/render.test.tsx]: Should be:
```
expect(expect(container.firstElementChild).toBeTruthy();
expect(container.firstElementChild!.className).toBe('date-time-container');
```

File: src/react/DateTime/__test__/render.test.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        locale="zh"
+        dateFormat="YYYY/MM"
+        timeFormat="HH:mm:ss"
+        isDisabled
+        isVisible
+      />
+    );
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('date-time-container');
+    }
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/render.test.tsx]: Should be:
```
expect(expect(container.firstElementChild).toBeTruthy();
expect(container.firstElementChild!.className).toBe('date-time-container');
```

File: src/react/DateTime/__test__/render.test.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        locale="zh"
+        dateFormat="EE/d/M/Y"
+        timeFormat="H:m"
+        isDisabled
+        isVisible
+      />
+    );
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('date-time-container');
+    }
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/render.test.tsx]: Should be:
```
expect(expect(container.firstElementChild).toBeTruthy();
expect(container.firstElementChild!.className).toBe('date-time-container');
```

Add expect for date format

File: src/react/DateTime/__test__/render.test.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      <DateTime
+        value={new Date()}
+        type="datetime"
+        locale="zh"
+        dateFormat="EEEE/d/M/Y"
+      />
+    );
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('date-time-container');
+    }
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/render.test.tsx]: Should be:
```
expect(expect(container.firstElementChild).toBeTruthy();
expect(container.firstElementChild!.className).toBe('date-time-container');
```

Add expect for date format

File: src/react/DateTime/__test__/render.test.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      <DateTime
+        value={new Date()}
+        type="datetime"
+        locale="zh"
+        dateFormat="calendartitle"
+      />
+    );
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('date-time-container');
+    }
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/render.test.tsx]: Should be:
```
expect(expect(container.firstElementChild).toBeTruthy();
expect(container.firstElementChild!.className).toBe('date-time-container');
```

Add expect for date format

File: src/react/DateTime/__test__/render.test.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    const {container} = render(<DateTime value={new Date()} isVisible={false} locale="en" />);
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('');
+    }
+  });
+  test('render with wrong props DateTime', () => {
+    // @ts-ignore
+    const {container} = render(<DateTime value={new Date()} isDisabled="false" type="kintone" />);
+    if (container.firstElementChild) {
+      expect(container.firstElementChild.className).toBe('date-time-container');
```
Comment (phongnhat19) [inline src/react/DateTime/__test__/render.test.tsx]: Add expect for empty div

File: src/js/DateTime/__test__/disable.test.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    // @ts-ignore
+    console.error.mockRestore();
+  });
+
+  test('disable & enable successfully DateTime', () => {
+    const datetime = new DateTime();
+    const dateTextInput = datetime.render().getElementsByTagName('input')[0];
+    fireEvent.click(dateTextInput, {target: {value: null}});
+    datetime.disable();
+    expect(datetime.render().getAttribute('disabled')).toBe('true');
```
Comment (phongnhat19) [inline src/js/DateTime/__test__/disable.test.ts]: Should expect the `disabled` attribute of the text input, not the `div` element.

File: src/js/DateTime/__test__/disable.test.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    datetime.enable();
+    expect(datetime.render().getAttribute('disabled')).toBeNull();
```
Comment (phongnhat19) [inline src/js/DateTime/__test__/disable.test.ts]: Should expect the `disabled` attribute of the text input, not the `div` element.

File: src/js/DateTime/__test__/onEvent.test.ts [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    fireEvent.keyDown(time, {keyCode: 100, target: {selectionStart: 1}});
+    fireEvent.keyDown(time, {keyCode: 48, target: {selectionStart: 1}});
+    fireEvent.blur(time);
+    fireEvent.blur(time, {relatedTarget: null});
+    // @ts-ignore
+    datetime._setTimeValueOnInput('kintone');
+    expect(true).toBeTruthy();
+  });
+
+  test('onEvent dateTextInput throws error DateTime', () => {
```
Comment (phongnhat19) [inline src/js/DateTime/__test__/onEvent.test.ts]: Change description for clear meaning.

File: src/js/DateTime/__test__/onEvent.test.ts [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    const datetime = new DateTime({type: 'time', value: new Date('Mon, 6 Jan 2020 08:2:00 GMT')});
+    const datetimeRender = datetime.render();
+    const timePicker = datetimeRender.getElementsByClassName('kuc-time-list-item');
+    const time = datetimeRender.getElementsByClassName('kuc-input-text text-input time')[0];
+    fireEvent.click(time, {target: {selectionStart: 4}});
+    fireEvent.click(timePicker[30]);
+    fireEvent.click(time, {target: {selectionStart: 3}});
+    fireEvent.click(timePicker[5]);
+    expect(true).toBeTruthy();
```
Comment (phongnhat19) [inline src/js/DateTime/__test__/onEvent.test.ts]: Incorrect `expect`

File: src/js/DateTime/__test__/Calendar.test.ts [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    expect(oldFirstDay.textContent).toBe('26');
+
+    const span = pickerContainer.getElementsByClassName('next calendar-button-control')[0];
+    fireEvent.click(span);
+    const newFirstDay = pickerContainer.getElementsByClassName('day')[0];
+    expect(newFirstDay.textContent).toBe('1');
+  });
+  test('onClick Today Button Calendar', () => {
+    const mockFn = jest.fn((date: Date | null) => {
+      const expectToday = new Date();
```
[the remark this answers is not in the document]
Comment (hokimcuc0202) [reply src/js/DateTime/__test__/Calendar.test.ts]: This case will be covered in line 61

**Pull request merged**